common/workunit/workunit.cpp

```diff
@@ -14018,23 +14018,55 @@ bool executeGraphOnLingeringThor(IConstWorkUnit &workunit, const char *graphName
 
 static void setResources(StringBuffer &jobYaml, const IConstWorkUnit *workunit, const char *process)
 {
+    Owned<IPropertyTree> cconfig;
+    
+    if (strcmp(process,"thormaster")==0)
+    {
+        dbglogXML(&queryComponentConfig());
+        cconfig.setown(queryComponentConfig().getPropTree("thorMasterResources"));
+    }
+    else if (strcmp(process,"thorslave")==0) 
+        cconfig.setown(queryComponentConfig().getPropTree("thorSlaveResources"));
+    else
+        cconfig.setown(queryComponentConfig().getPropTree("childContainersResources"));
```

Comment:
Feels like the selection of a section in values etc. and generation of resources should mainly be done by the helm chart templates/helper.
Substitiouon of any overrides can still be achieved in a generic way I think, when processing a jobspec, by matching the locating matching process type properties, parsing them and setting them into the generated and loaded yaml resource section.